fastlane/fastlane
# [sigh] Fix resign for widget (or other plugins)

🔑


### Checklist

### Motivation and Context

When resigning an ios app which has a widget without an extra app entitlement and using `--use_app_entitlements`, the resigning fails with the following error. I don't know if there are actually ios apps where the widget has its own app entitlements file.
```
Executable=/.../_floatsignTemp/Payload/myApp.app/PlugIns/Widget.appex/Widget
...
_floatsignTemp/patchedEntitlements: unrecognized blob type (accepting blindly)
_floatsignTemp/patchedEntitlements: invalid length in entitlement blob
Encountered an error, aborting!
```
This error occurs inside `sigh/lib/assets/resign.sh` because it expects an non empty app entitlements file. The main app has one. But the Widget (which is inside the plugins folder within the ipa) does not.



### Description

More specifically the error occurs on on line 829 (https://github.com/fastlane/fastlane/blob/master/sigh/lib/assets/resign.sh#L829), because of accumulated error / failed attempts to parse the app entitlements.
I moved the parts which depend on the app entitlements into an _if_, so they are not executed if no app entitlements file is currently present. The patched entitlements file however is still generated.


I tested the changes on my app, which failed to be resigned. Now the resigning works and the app can be successfully deployed (to testers via ad-hoc).

### Testing Steps


I am sorry I can not provide an example app. I am also not sure how to add a test to fastlane which tests the resigning of such a condition. This would be really good to avoid this feature to be broken again in the future. It is also not enough to only check if the resigning ended successfully. One also needs to deploy the app to an actual device to ensure the structure and codesigning is valid.
